doc/application/application.rst

```diff
@@ -851,13 +851,15 @@ Kconfig Configuration
 =====================
 
 The initial configuration for an application is produced by merging
-configuration settings from two sources:
+configuration settings from three sources:
 
 1. A :makevar:`BOARD`-specific configuration file, stored in
    :file:`boards/ARCHITECTURE/BOARD/BOARD_defconfig` in the Zephyr base
    directory.
 
-2. One or more application-specific configuration files.
+2. Any CMakeCache entries entries that are prefixed with :makevar:`CONFIG_`.
```

Comment:
It would be nice to have an explanation of what the CMake cache is and how stuff ends up in it, for people coming from `make` and the like.

Reply:
Maybe add something like this:

    Cmake uses a CMakeCache.txt file as persistent key/value string storage
    used to cache values between runs, including compile and build options, 
    paths to library dependencies,  or whether an optional component is to be
    built at all.  This cache file is created when Cmake is run in an empty
    build folder.

Reply:
Could work. Does that mean that if you do e.g. `cmake CONFIG_FOO=y ...` and then just `cmake ...`, `CONFIG_FOO=y` will be remembered the second time?

Guess it's kinda implied, but could maybe be phrased in those terms as well.

Reply:
I just learned about `CMakeCache.txt` btw, so thanks for that. ;)

Reply:
> [..] Does that mean that if you do e.g. cmake CONFIG_FOO=y .. and then just cmake .., CONFIG_FOO=y will be remembered the second time? [..]  it's kinda implied, but could maybe be phrased in those terms as well.

I think it would be nice to clarify this too. As it is not too obvious that is the case.